docs/topics/multiplatform/multiplatform-compatibility-guide.md

```diff
@@ -910,6 +912,41 @@ Here's the planned deprecation cycle:
   with "unresolved reference" errors, and binaries (for example, Gradle plugins) might fail with linkage errors
   unless recompiled against the latest versions of the Kotlin Gradle plugin
 
+<anchor name="target-shortcuts-deprecation"/>
+## Deprecated Apple target shortcuts
+
+**What's changed?**
+
+We're deprecating `ios()`, `watchos()`, and `tvos()` target shortcuts in Kotlin Multiplatform DSL. They were designed to
+partially create a source set hierarchy for Apple targets. However, they proved to be difficult to expand and sometimes confusing.
+
+For example, the `ios()` shortcut created both the `iosArm64` and `iosX64` targets but didn't include the `iosSimulatorArm64`
+target, which is necessary when working on hosts with Apple M chips. However, changing this shortcut was hard to implement
+and could cause issues in existing user projects.
+
+**What's the best practice now?**
+
+The Kotlin Gradle plugin now provides a built-in hierarchy template. Since Kotlin 1.9.20, it's enabled by default
+and contains predefined intermediate source sets for popular use cases.
+
+The difference from shortcuts is that you need to explicitly specify the set of targets first, and the plugin then
+automatically sets up intermediate source sets based on this list.
+
+For example, suppose you have `iosArm64` and `iosSimulatorArm64` targets in your project. In that case, the plugin
+automatically creates the `iosMain` and `iosTest` intermediate source sets, and for the `iosArm64` and `macosArm64`
+targets, it creates the `appleMain` and `appleTest` source sets.
```

Comment:
I don't think I understand where the `macosArm64` target comes from, if we didn't suppose that the project has it from the beginning. Is "`iosArm64` and `iosSimulatorArm64`" one example and "`iosArm64` and `macosArm64`" another?

Reply:
Yes, another example to show that different intermediate source sets are created, I should probably split it into two senetences